x-pack/plugins/security_solution/public/management/pages/policy/view/ingest_manager_integration/endpoint_policy_create_extension/endpoint_policy_create_extension.tsx

```diff
@@ -102,7 +101,7 @@ export const EndpointPolicyCreateExtension = memo<PackagePolicyCreateExtensionCo
     // only during 1st component render (thus why the eslint disabled rule below).
     // Default values for config are endpoint + NGAV
     useEffect(() => {
-      if (initialRender.current) {
+      if (newPolicy.inputs.length === 0) {
```

Comment:
Got curious, can a newPolicy have empty `inputs` array on subsequent renders? Just want to be sure that `newPolicy.inputs.length === 0` means first render and this code will not be executed more than once.

Reply:
It shouldn't have empty inputs (at least not for endpoint) on subsequent renders. And yup the goal was to have the inputs.length be equivalent to an initial render, since inputs is empty for the endpoint package until we populate it.